### ISSUE TITLE
Reconsider dial box implementation

### ISSUE THREAD

**Issue by terasakisatoshi** [AtelierArith/TerminalClock.jl#17]
To achieve #16, I need to reconsider the design of `src/TerminalClock.jl`. 

Here are my thoughts:

1. Create a Julia script below:

```julia
using TOML
using Configurations

abstract type FontSize end
struct Small <: FontSize end
struct Medium <: FontSize end
struct Large <: FontSize end

# Actually we do not have to add parametric type.
@option "dial" struct Dial{T<:FontSize}
    one::Vector{String}
    two::Vector{String}
    #three::Vector{String}
    #four::Vector{String}
    #five::Vector{String}
    #six::Vector{String}
    #seven::Vector{String}
    #eight::Vector{String}
    #nine::Vector{String}
    #zero::Vector{String}
    colon::Vector{String}
end

function n2d(dial::Dial, n::Int)
    n == 0 && return getfield(dial, :zero)
    n == 1 && return getfield(dial, :one)
    n == 2 && return getfield(dial, :two)
    n == 3 && return getfield(dial, :three)
    n == 4 && return getfield(dial, :four)
    n == 5 && return getfield(dial, :five)
    n == 6 && return getfield(dial, :six)
    n == 7 && return getfield(dial, :seven)
    n == 8 && return getfield(dial, :eight)
    n == 9 && return getfield(dial, :nine)
    DomainError("n should satisfy 0≤ n ≤9, actual $n")
end

colon(dial::Dial) = getfield(dial, :colon)

# example

file = "MyDial.toml"
toml = TOML.parsefile(file)

# from_dict(Dial, toml[...]) is fine
d_small = from_dict(Dial{Small}, toml[string(nameof(Small))])
d_medium = from_dict(Dial{Medium}, toml[string(nameof(Medium))])
d_large = from_dict(Dial{Large}, toml[string(nameof(Large))])

# do something e.g. clock, stopwatch, countdown

@show n2d(d_small, 1)
#=
3-element Vector{String}:
 "smallone"
 "++++++++"
 "--------"
=#
@show colon(d_medium)
#=
1-element Vector{String}:
 "mediumcolon"
=#
```

where `MyDial.toml` is something like

```toml
[Large]

# don't get me, wrong this is a lazy implementation 

one = [
"largeone",
"++++++++",
"--------",
]

two = [
"largetwo",
"++++++++",
"--------",
]

colon = [
"largecolon",
]

[Medium]

one = [
"mediumone",
"++++++++",
"--------",
]

two = [
"mediumtwo",
"+++++++++",
"---------",
]

colon = [
"mediumcolon",
]

[Small]

one = [
"smallone",
"++++++++",
"--------",
]

two = [
"smalltwo",
"++++++++",
"--------",
]

colon = [
"smallcolon"
]
```

**Comment by terasakisatoshi** [comment]
I would like to apply this technique to our package.

https://github.com/JuliaLang/IJulia.jl/blob/8710cd4bc1adabfacb8d9caec56332e36daa1f90/deps/build.jl#L14-L16

As for the environment variable for our package, here are my thoughts:

```julia
julia> ENV["JULIA_TERMINALCLOCK_DIALS"]="path/to/MyDials.toml"
julia> using Pkg; Pkg.build("TerminalClock")
julia> using TerminalClock; clock() # will load ~/.julia/prefs/TerminalClock/MyDials.toml" and diplay dials you created
```

🤔

**Comment by terasakisatoshi** [comment]
closed via #21 and #22
